### PR TITLE
[ENH] Refactor BOLD workflows

### PR DESCRIPTION
**Pull request by oesteban**
Create a `bold` module that makes it more manageable contributing to FMRIPREP.

Generation of API for workflows has been revised (refs. #615).

### CODE REVIEW

File: fmriprep/workflows/bold/base.py [hunk 1]
```diff
@@ -0,0 +1,785 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Orchestrating the BOLD-preprocessing workflow
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. autofunction:: init_func_preproc_wf
+.. autofunction:: init_func_reports_wf
+.. autofunction:: init_func_derivatives_wf
```
Comment (effigies) [inline fmriprep/workflows/bold/base.py]: Do you really want to include reports and derivatives in the API? I think I skipped them. But I suppose no harm, either.

File: fmriprep/workflows/bold/base.py [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+            (bold_reference_wf, bold_hmc_wf, [('outputnode.bold_file', 'inputnode.bold_file')])])
+
+    # Cases:
+    # fmaps | use_syn | force_syn  |  ACTION
+    # ----------------------------------------------
+    #   T   |    *    |     T      | Fieldmaps + SyN
+    #   T   |    *    |     F      | Fieldmaps
+    #   F   |    *    |     T      | SyN
+    #   F   |    T    |     F      | SyN
+    #   F   |    F    |     F      | HMC only
```
Comment (effigies) [inline fmriprep/workflows/bold/base.py]: If you're going to do this, should we make the choice of fieldmap unwarping and  SyN-based unwarping part of the fieldmaps workflows? Though, that may make more sense as a separate PR, as this one is already pretty big. Either way.

File: fmriprep/workflows/bold/base.py (hunk 1 above)
Comment (oesteban) [reply fmriprep/workflows/bold/base.py]: It generates two entries with one-liner description. May even help to newcomers.

File: fmriprep/workflows/bold/base.py (hunk 2 above)
Comment (oesteban) [reply fmriprep/workflows/bold/base.py]: Yes - the SyN-based workflow should be attached the same way we are currently doing with the fieldmap decision maker in `workflows.fieldmap.base` if we want to be consistent. But better in a different PR where that's the only target.

**Comment by oesteban** [comment]
Seems like everything is in place. Gong ahead.
